### PR TITLE
tso field  from string to object

### PR DESCRIPTION
**Pull request by ChamseddineBhd**

### CODE REVIEW

File: src/main/java/org/gridsuite/merge/orchestrator/server/repositories/MergeRepository.java [hunk 1]
```diff
@@ -24,6 +24,9 @@
     @Query("SELECT * FROM merge WHERE process = :process")
     List<MergeEntity> findByProcess(@Param("process") String process);
 
+    @Query("DELETE FROM merge WHERE process = :process")
+    void deleteByProcess(@Param("process") String process);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/repositories/MergeRepository.java:28]: Is the @Param annotation required ?

File: src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java [hunk 2]
```diff
@@ -94,15 +93,16 @@ public MergeOrchestratorService(CaseFetcherService caseFetchService,
         return f -> f.log(CATEGORY_BROKER_INPUT, Level.FINE).subscribe(this::consume);
     }
 
-    private boolean checkTso(List<String> tsos, String tso, String format, String businessProcess) {
-        return tsos.contains(tso) && StringUtils.equals(format, ACCEPTED_FORMAT) && StringUtils.isNotEmpty(businessProcess);
+    private boolean checkTso(List<Tso> tsos, String tso, String format, String businessProcess) {
+        return tsos.stream().anyMatch(ts -> ts.getSourcingActor().equals(tso) || ts.getAlternativeSourcingActor().equals(tso))
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java:97]: Maybe a method for matching the sourcingActor or alternativeSourcingActor ?

File: src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ public void consume(Message<String> message) {
                 ZonedDateTime dateTime = ZonedDateTime.parse(date);
 
                 for (ProcessConfig processConfig : processConfigs) {
-                    if (processConfig.getTsos().contains(tso)) {
+                    if (processConfig.getTsos().stream().anyMatch(ts -> ts.getSourcingActor().equals(tso) || ts.getAlternativeSourcingActor().equals(tso))) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java:120]: Calling the method defined above

File: src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java [hunk 4]
```diff
@@ -144,7 +144,7 @@ public void consume(Message<String> message) {
     }
 
     void merge(ProcessConfig processConfig, ZonedDateTime dateTime, String date, String tso, boolean valid, UUID networkUuid) {
-        if (processConfig.getTsos().contains(tso)) {
+        if (processConfig.getTsos().stream().anyMatch(ts -> ts.getSourcingActor().equals(tso) || ts.getAlternativeSourcingActor().equals(tso))) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java:147]: Calling the method defined above

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/merge/orchestrator/server/repositories/MergeRepository.java (hunk 1 above)
Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/merge/orchestrator/server/repositories/MergeRepository.java:28]: yes it's optional

**Review by FranckLecuyer — APPROVED** [review]

**Review by jonenst — CHANGES_REQUESTED** [review]
split PR in multiple PRs for different things

**Review by jonenst — APPROVED** [review]

**Pull request merged**